modules/core/src/main/java/org/apache/ignite/cache/QueryEntity.java

```diff
@@ -780,7 +780,7 @@ private static void processAnnotationsInClass(boolean key, Class<?> cls, QueryEn
         @Nullable QueryEntityClassProperty parent) {
         if (U.isJdk(cls) || QueryUtils.isGeometryClass(cls)) {
             if (parent == null && !key && QueryUtils.isSqlType(cls)) { // We have to index primitive _val.
-                String idxName = cls.getSimpleName() + "_" + QueryUtils.VAL_FIELD_NAME + "_idx";
+                String idxName = QueryUtils.indexName(cls.getSimpleName(), QueryUtils.VAL_FIELD_NAME);
```

Comment:
1. Local method is enough
2. Looks like redundant check at all

Reply:
Fixed